compiler/modules/types/mapping.bal

```diff
@@ -225,7 +225,7 @@ function mappingInhabited(Context cx, TempMappingSubtype pos, Conjunction? negLi
         }
         foreach var { name, index1, type1: posType, type2: negType } in pairing {
             SemType d = diff(posType, negType);
-            if !isEmpty(cx, d) {
+            if !isEmpty(cx, d) || index1 is () {
```

Comment:
This doesn't look like a good fix to me. This might insert a named field with a type that is empty, which is generally not allowed.

Reply:
This is almost right. See https://github.com/ballerina-platform/nballerina/issues/1049#issuecomment-1197583882

But it's not doing quite the right thing when `index1 == ()`.

Reply:
fixed